task/acs-image-check/0.1/acs-image-check.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: acs-image-check
+spec:
+  description: Policy check an image with StackRox/RHACS This tasks allows you to
+    check an image against build-time policies and apply enforcement to fail builds.
+    It's a companion to the stackrox-image-scan task, which returns full vulnerability
+    scan results for an image.
+  params:
+  - name: rox-central-endpoint
+    type: string
+    description: |
+      The address:port tuple for StackRox Central.
+      For example: rox.stackrox.io:443
+  - name: rox-api-token-secret-name
+    type: string
+    description: |
+      Secret containing the StackRox API token with CI permissions under rox-api-token key
+  - name: image
+    type: string
+    description: |
+      Full name of image to scan (example -- gcr.io/rox/sample:5.0-rc1)
+  - name: insecure-skip-tls-verify
+    type: string
+    default: "false"
+    description: |
+      When set to `"true"`, skip verifying the TLS certs of the Central
+      endpoint.  Defaults to `"false"`.
+  - name: image-digest
+    type: string
+    description: |
+      Digest of the image
+  results:
+  - description: Output of `roxctl image check`
+    name: check_output
+  volumes:
+    - name: rox-secret
+      secret:
+        secretName: $(params.rox-api-token-secret-name)
+        optional: true
+  steps:
+  - name: rox-image-check
+    image: registry.access.redhat.com/ubi8/ubi-minimal
+    volumeMounts:
+      - name: rox-secret
+        mountPath: /rox-secret
+    script: |
+      #!/usr/bin/env bash
+      set +x
+
+      # Check if rox API token is configured
+      if test -f /rox-secret/rox-api-token ; then
+        export ROX_API_TOKEN=$(</rox-secret/rox-api-token)
+      else
+        echo "rox API token is not set, demo will exit with success"
+        echo "TODO: configure the pipeline to have access to ROXCTL"
+        exit 0
+      fi
+
+      echo "Using rox central endpoint $(params.rox-central-endpoint)"
+
+      echo "Download roxctl cli"
+      IMAGE=$(params.image)@$(params.image-digest)
+      curl -s -k -L -H "Authorization: Bearer $ROX_API_TOKEN" \
```

Comment:
If we need to pull the CLI from a remote location, could we not use `-k` (`--insecure`)?
Could we pull from mirror.openshift.com [as documented](https://docs.openshift.com/acs/4.3/cli/installing-the-roxctl-cli.html) instead? Ideally an image with the `roxctl` would be the best option here, is there an image we could use instead?

Reply:
Downloading is done mainly due to the following reasons:
 - The image containing `roxctl` is stored in `registry.redhat.com` which requires subscription to pull.
 - `roxctl` version in the image might be different from ACS version used by consumer.

Regarding insecure, I agree, we have to have it controller. Probably by `insecure-skip-tls-verify` parameter.

Reply:
> The image containing roxctl is stored in registry.redhat.com which requires subscription to pull.

There are ways to solve this, e.g. document that you need a subscription... This is not a reason not to use that registry IMO.

> roxctl version in the image might be different from ACS version used by consumer.

Then provide different version of the Task for different ACS versions?

Reply:
I see there is quay.io/repository/stackrox-io/roxctl, could that be used? In principal if the version matches the ACS server version? Perhaps using that from the start and then giving a reasonable window of supported CLI vs server version could be an option? 

Another idea, we could have many versions of the Task here matching the (major?) versions of ACS/stackrox using the compatible CLI.

I would not like to see any insecure fetching, that could allow for man in the middle attacks.

Reply:
> There are ways to solve this, e.g. document that you need a subscription... This is not a reason not to use that registry IMO.

If we use images that require subscription to pull in the RHTAP pipeline, it would mean, that whole RHTAP pipeline wouldn't run without subscription. I'm not sure that it's a desired "feature" for the RHTAP pipeline.
cc @jduimovich @arewm

Reply:
> Another idea, we could have many versions of the Task 

I'd really like to avoid this.
Otherwise, we only pile up a lot of tasks we'll have to maintain.

Reply:
FWIW - this same code, with insecure download -k code is found in the "official" task https://hub.tekton.dev/tekton/task/stackrox-image-scan. 
We need to work with the stackrox team to resolve the one correct way to get a binary for roxctl. Not sure why they control access to the CLI, as you can only use it with an installed ACS server.

Reply:
> If we use images that require subscription to pull in the RHTAP pipeline, it would mean, that whole RHTAP pipeline wouldn't run without subscription
Can we just reference the image in the task itself such that we can protect against the lack of valid credentials?

Users should have credentials for registry.redhat.io to use, but would this require _each_ user to specify the credentials (i.e. not the PEs)?

Reply:
@arewm this would require pull secret in the pipeline namespace (user's namespace).
Sharing the same token is not secure, so each user must provide own secret.